server/task_executor_database_create.go

```diff
@@ -103,21 +103,28 @@ func (exec *DatabaseCreateTaskExecutor) RunOnce(ctx context.Context, server *Ser
 	// 1. Assign the proper project to the newly created database. Otherwise, the periodic schema
 	// sync will place the synced db into the default project.
 	// 2. Allow user to see the created database right away.
-	databaseCreate := &api.DatabaseCreate{
-		CreatorID:     api.SystemBotID,
-		ProjectID:     payload.ProjectID,
-		InstanceID:    task.InstanceID,
-		EnvironmentID: instance.EnvironmentID,
-		Name:          payload.DatabaseName,
-		CharacterSet:  payload.CharacterSet,
-		Collation:     payload.Collation,
-		Labels:        &payload.Labels,
-		SchemaVersion: payload.SchemaVersion,
-	}
-	database, err := server.store.CreateDatabase(ctx, databaseCreate)
+	database, err := server.store.GetDatabase(ctx, &api.DatabaseFind{InstanceID: &task.InstanceID, Name: &payload.DatabaseName})
 	if err != nil {
 		return true, nil, err
 	}
+	if database == nil {
```

Comment:
This does not prevent the project from being incorrectly stored in the Default project.

Reply:
I think we should follow this strategy:
1. check if the current project id is as expected
2. If not, update the project id of the current database to match the expectation

Reply:
This is intended. The database already exists in the 1) default project or 2) a different project. When inconsistency happens, it's better to fail the database creation issue and let DBAs resolve the conflict by moving the database to the right project.

Reply:
Maybe we can check if there is a create database issue being executed while synchronizing the database in the background, and if so, skip the synchronization of that database for now

Reply:
Done.